lib/jxl/dec_ans.h

```diff
@@ -302,53 +303,62 @@ class ANSSymbolReader {
   }
 
   // Takes a *clustered* idx.
-  size_t ReadHybridUintClustered(size_t ctx, BitReader* JXL_RESTRICT br) {
-    if (JXL_UNLIKELY(num_to_copy_ > 0)) {
-      size_t ret = lz77_window_[(copy_pos_++) & kWindowMask];
-      num_to_copy_--;
-      lz77_window_[(num_decoded_++) & kWindowMask] = ret;
-      return ret;
+  template <bool uses_lz77>
+  JXL_INLINE size_t ReadHybridUintClustered(size_t ctx,
```

Comment:
How do the speed and size results change without this JXL_INLINE?

Reply:
Most of the both comes from that. It's not ideal that it gets inlined everywhere now, also in the not hot parts. Is there a way to do an inlining macro on the call site rather than on the function definition?